server/svix-server/src/webhook.rs

```diff
@@ -0,0 +1,319 @@
+// SPDX-FileCopyrightText: © 2022 Svix Authors
+// SPDX-License-Identifier: MIT
+
+use http::{HeaderMap, HeaderValue};
+use time::OffsetDateTime;
+
+#[derive(thiserror::Error, Debug)]
+pub enum WebhookError {
+    #[error("failed to parse timestamp")]
+    InvalidTimestamp,
+
+    #[error("invalid secret")]
+    InvalidSecret(#[from] base64::DecodeError),
+
+    #[error("invalid header {0}")]
+    InvalidHeader(String),
+
+    #[error("signature timestamp too old")]
+    TimestampTooOldError,
+
+    #[error("signature timestamp too far in future")]
+    FutureTimestampError,
+
+    #[error("missing header {0}")]
+    MissingHeader(String),
+
+    #[error("signature invalid")]
+    InvalidSignature,
+
+    #[error("payload invalid")]
+    InvalidPayload,
+}
+
+pub struct Webhook {
+    key: Vec<u8>,
+}
+
+const PREFIX: &str = "whsec_";
+const SVIX_MSG_ID_KEY: &str = "svix-id";
+const SVIX_MSG_SIGNATURE_KEY: &str = "svix-signature";
+const SVIX_MSG_TIMESTAMP_KEY: &str = "svix-timestamp";
+const UNBRANDED_MSG_ID_KEY: &str = "webhook-id";
+const UNBRANDED_MSG_SIGNATURE_KEY: &str = "webhook-signature";
+const UNBRANDED_MSG_TIMESTAMP_KEY: &str = "webhook-timestamp";
+const ID: &str = "id";
```

Comment:
Don't make it a const (the error messages). It's used once and should only be used once. Splitting it like this is confusing.